pennylane/_queuing_context.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :class:`QueuingContext` abstract base class.
+"""
+import abc
+
+
+class QueuingContext(abc.ABC):
+    """Abstract base class for classes that exposes a queue for Operations.
+    """
+
+    _active_contexts = []
+    """The list of contexts that are currently active."""
```

Comment:
This is a big change right? Now, multiple contexts can be available at any one time? Probably doesn't make a difference except for when parallel/threading is used

Reply:
Yes, but this is basically inteded as 1 qnode + X operation recorders. 

Parallel construction of QNodes isn't possible with PL's Operation queue structure anyways, so this is not deteriorating at this end either.

Reply:
Ah, that makes sense